### PR TITLE
Enable debug flag for firewalld

### PR DESCRIPTION
**Pull request by foursixnine**
Changes to iptables over nft come from: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12347

VR:
- [opensuse-15.3-JeOS-for-kvm-and-xen-x86_64-Build9.13-jeos-extra@64bit_virtio-2G](phobos.qa.suse.de/tests/3857473)
- [ sle-15-SP3-Online-x86_64-Build176.5-extra_tests_textmode@64bit ](http://phobos.qa.suse.de/tests/3857470)
- [opensuse-Tumbleweed-DVD-x86_64-Build20210418-extra_tests_textmode@64bit](http://phobos.qa.suse.de/tests/3857471)
- [ opensuse-15.3-DVD-x86_64-Build134.3-extra_tests_textmode@64bit](http://phobos.qa.suse.de/tests/3857467)
- [sle-12-SP5-Server-DVD-Incidents-x86_64](https://openqa.suse.de/tests/5858962)

### CODE REVIEW

File: tests/console/firewalld.pm [hunk 1]
```diff
@@ -18,6 +18,10 @@ use testapi;
 use utils qw(systemctl zypper_call);
 use version_utils qw(is_sle is_leap is_tumbleweed);
 
+sub has_iptables {
```
Comment (asdil12) [inline tests/console/firewalld.pm:21]: Fabian suggested calling it `uses_iptables`

Comment (foursixnine) [reply tests/console/firewalld.pm]: done

File: tests/console/firewalld.pm [hunk 2]
```diff
@@ -233,15 +232,15 @@ sub test7 {
     if (is_sle('15-SP3+') || is_leap('15.3+')) {
```
Comment (asdil12) [inline tests/console/firewalld.pm:232]: This if branch needs to be removed and the next `elsif` needs to be the `if:

https://openqa.opensuse.org/tests/1705721#step/firewalld/208

**Review by asdil12 — CHANGES_REQUESTED** [review]

File: tests/console/firewalld.pm (hunk 2 above)
Comment (foursixnine) [reply tests/console/firewalld.pm]: @asdil12 done

File: tests/console/firewalld.pm [hunk 3]
```diff
@@ -8,7 +8,7 @@
 # without any warranty.
 
 # Package: firewalld
-# Summary: Test FirewallD basic usage
+# Summary: Test FirewallD basic usage, including NFT tables
```
Comment (Vogtinator) [inline tests/console/firewalld.pm:11]: LCD display!

```suggestion
# Summary: Test FirewallD basic usage, including nftables/iptables
```

Comment (foursixnine) [reply tests/console/firewalld.pm:11]: https://knownorigin.io/gallery/254225-i-don-t-know-look-at-it-ix!

**Review by Vogtinator — CHANGES_REQUESTED** [review]

File: tests/console/firewalld.pm [hunk 4]
```diff
@@ -18,23 +18,22 @@ use testapi;
...
+    return is_sle('<15-SP3') || is_leap('<15.3');
+}
+
 # Check Service State, enable it if necessary, set default zone to public
 sub pre_test {
     zypper_call('in firewalld');
     zypper_call('info firewalld');
     record_info 'Check Service State';
-    assert_script_run("if ! systemctl is-active -q firewalld; then systemctl start firewalld; fi");
+    script_run('echo "FIREWALLD_ARGS=--debug" > /etc/sysconfig/firewalld');
```
[the remark this answers is not in the document]
Comment (foursixnine) [reply tests/console/firewalld.pm:30]: By default that file only has couple of lines, and the `FIREWALLD_ARGS` is still empty, I prefer to control that only debug flag is enabled this way.

File: tests/console/firewalld.pm [hunk 5]
```diff
@@ -18,23 +18,22 @@ use testapi;
...
+    return is_sle('<15-SP3') || is_leap('<15.3');
+}
+
 # Check Service State, enable it if necessary, set default zone to public
 sub pre_test {
     zypper_call('in firewalld');
     zypper_call('info firewalld');
     record_info 'Check Service State';
-    assert_script_run("if ! systemctl is-active -q firewalld; then systemctl start firewalld; fi");
+    script_run('echo "FIREWALLD_ARGS=--debug" >> /etc/sysconfig/firewalld');
```
Comment (Vogtinator) [inline tests/console/firewalld.pm:30]: Does systemd's `EnvironmentFile` support overwriting like this? I suggest using sed or awk instead

**Review by asdil12 — APPROVED** [review]

File: tests/console/firewalld.pm (hunk 5 above)
Comment (foursixnine) [reply tests/console/firewalld.pm:30]: > Does systemd's `EnvironmentFile` support overwriting like this? I suggest using sed or awk instead

Yeah, it does phobos.qa.suse.de/tests/3857474 :) at least no complains from openQA run, and enabling it on my machine: https://asciinema.org/a/JqxqBglPBNV3HLnX0UQYbOajG

Comment (Vogtinator) [reply tests/console/firewalld.pm:30]: Found it in the systemd.exec man page as well:

> If the same variable is set twice from these files, the files will be read in the order they are specified and the later setting will override the earlier setting.

Though I'm not sure whether that's about several files or several statements in the same file...

**Review by Vogtinator — APPROVED** [review]

**Pull request merged**